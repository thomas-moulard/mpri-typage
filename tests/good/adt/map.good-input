type list = 
  Nil
| Cons of int * list
end

val rec map f l =
  match l with
    | Nil -> Nil
    | Cons x xs -> Cons (f x) (map f xs) 
  end

val l = Cons 1 (Cons 2 Nil)

val zl = map (fun x. 0) l
