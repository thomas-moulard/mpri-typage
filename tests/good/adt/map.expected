Nil = Nil

Cons = Cons

map = fun f.fun l.match l with 
    | Nil -> Nil
    | Cons x xs -> Cons (f x) ((let rec map1 f1 l1 =
      match l1 with 
          | Nil -> Nil
          | Cons x1 xs1 -> ((Cons (f1 x1)) ((map1 f1) xs1))
      end

    in
map1 f) xs)
end


l = Cons 1 ((Cons 2) Nil)

zl = Cons 0 ((Cons 0) Nil)

zl : list
l : list
map : ((int -> int) -> (list -> list))
Cons : (int -> (list -> list))
Nil : list
